knative-operator/pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -51,6 +57,40 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	resources, err := kourier.Resources(ingressNamespace)
+	if err != nil {
+		return err
+	}
+	// Watch Kourier resources.
+	for _, u := range resources {
+		g := u.GroupVersionKind()
+		gvk := schema.GroupVersionKind{Group: g.Group, Kind: g.Kind, Version: g.Version}
+		u.SetGroupVersionKind(gvk)
+		name := u.GetName()
+		namespace := u.GetNamespace()
+		err = c.Watch(&source.Kind{Type: &u}, &handler.EnqueueRequestsFromMapFunc{
+			ToRequests: handler.ToRequestsFunc(func(obj handler.MapObject) []reconcile.Request {
+				requests := []reconcile.Request{}
+				requests = append(requests, reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name:      name,
+						Namespace: namespace,
```

Comment:
This appends the wrong resources.

Instead, we need to append `knative-serving/knative-serving` in any case today. In theory, we need to build a relationship between the kourier resources and the KnativeServing. We could do that today by dropping the `-ingress` suffix from the namespace and then list a KnativeServing in the resulting namespace.

Today though, I think we can live with just doing this:

```golang
return []reconcile.Request{{
  NamespacedName: types.NamespacedName{
    Namespace: "knative-serving",
    Name: "knative-serving",
  }
}}
```

Potentially taking the strings as constants from somewhere.

Reply:
We'll also need to filter all resources aggressively. As it stands, we're watching all deployments, secrets, configmaps etc. in the cluster. We need to filter those down to just the namespace/name pairs we'd like to watch.